src/Sylius/Behat/spec/Context/Ui/Admin/ManagingCountryContextSpec.php

```diff
@@ -0,0 +1,94 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace spec\Sylius\Behat\Context\Ui\Admin;
+
+use Behat\Behat\Context\Context;
+use PhpSpec\Exception\Example\NotEqualException;
+use PhpSpec\ObjectBehavior;
+use Sylius\Behat\Page\Admin\Country\CreatePageInterface;
+use Sylius\Behat\Page\Admin\Crud\IndexPageInterface;
+use Sylius\Component\Addressing\Model\CountryInterface;
+use Sylius\Component\Core\Test\Services\SharedStorageInterface;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+class ManagingCountryContextSpec extends ObjectBehavior
+{
+    function let(IndexPageInterface $countryIndexPage, CreatePageInterface $countryCreatePage)
+    {
+        $this->beConstructedWith($countryIndexPage, $countryCreatePage);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Behat\Context\Ui\Admin\ManagingCountryContext');
+    }
+
+    function it_is_context()
+    {
+        $this->shouldImplement(Context::class);
+    }
+
+    function it_opens_country_creation_page(CreatePageInterface $countryCreatePage)
+    {
+        $countryCreatePage->open()->shouldBeCalled();
+
+        $this->iWantToCreateNewCountry();
+    }
+
+    function it_fills_name_in_creation_form_and_saves_it_in_shared_storage(CreatePageInterface $countryCreatePage)
+    {
+        $countryCreatePage->chooseName('France')->shouldBeCalled();
+
+        $this->iChoose('France');
+    }
+
+    function it_presses_create_button_on_creation_page(CreatePageInterface $countryCreatePage)
+    {
+        $countryCreatePage->create()->shouldBeCalled();
+
+        $this->iAddIt();
+    }
+
+    function it_asserts_that_successful_message_appears(IndexPageInterface $countryIndexPage)
+    {
+        $countryIndexPage->hasSuccessMessage()->willReturn(true);
+        $countryIndexPage->isSuccessfullyCreated()->willReturn(true);
+
+        $this->iShouldBeNotifiedAboutSuccess();
+    }
+
+    function it_throws_not_equal_exception_if_message_is_not_successful(IndexPageInterface $countryIndexPage)
+    {
+        $countryIndexPage->hasSuccessMessage()->willReturn(false);
+        $countryIndexPage->isSuccessfullyCreated()->willReturn(true);
+
+        $this->shouldThrow(NotEqualException::class)->during('iShouldBeNotifiedAboutSuccess');
+    }
+
+    function it_throws_not_equal_exception_if_successful_message_does_not_appear(IndexPageInterface $countryIndexPage)
```

Comment:
`...if_successfully_created_message...` or  something along those lines, because the success flash does appear it's just not the one we were anticipating.